src/platform/unix/mod.rs

```diff
@@ -660,10 +658,68 @@ fn make_socket_lingering(sockfd: c_int) -> Result<(),UnixError> {
     Ok(())
 }
 
+struct BackingStore {
+    fd: c_int
+}
+
+impl BackingStore {
+    pub fn new(length: usize) -> BackingStore {
+        let count = GLOBAL_SHM_COUNT.fetch_add(1, Ordering::Relaxed);
+        let timestamp = UNIX_EPOCH.elapsed().unwrap();
+        let pid = unsafe { libc::getpid() };
```

Comment:
Nah, I actually only mentioned using `lazy_static` for the counter before -- not realising that an atomic is the more obvious choice for that one...